### PR TITLE
Azure integration

### LINKED ISSUE #1162: Add Azure as supported identity provider
Need to support Azure as identity provider.
Admin role should be granted by username/email and azure roles.

### PR DESCRIPTION
**Pull request by adyachenkoprov**
closes #1162 Add Azure as supported identity provider

### CODE REVIEW

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/config/OAuthSecurityConfiguration.java [hunk 1]
```diff
@@ -69,6 +70,7 @@
 @EnableReactiveMethodSecurity
 @EnableWebFluxSecurity
 @RequiredArgsConstructor
+@Slf4j
```
Comment (DementevNikita) [inline odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/config/OAuthSecurityConfiguration.java:73]: seems unnecessary here, could you please revert that?

File: odd-platform-api/src/main/resources/application.yml [hunk 2]
```diff
@@ -26,6 +26,39 @@ auth:
   ingestion:
     filter:
       enabled: false
+
+#  oauth2:
```
Comment (DementevNikita) [inline odd-platform-api/src/main/resources/application.yml:30]: @adyachenkoprov @damirabdul
gents, let's add new provider docs to the GitBook right away, if needed

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/config/OAuthSecurityConfiguration.java (hunk 1 above)
Comment (adyachenkoprov) [reply odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/config/OAuthSecurityConfiguration.java]: removed

File: odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/auth/handler/impl/AzureUserHandler.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        boolean isAdmin = false;
+
+        if (CollectionUtils.isNotEmpty(provider.getAdminPrincipals())) {
+            final String adminPrincipalAttribute;
+            if (StringUtils.isNotEmpty(provider.getAdminAttribute())) {
+                adminPrincipalAttribute = provider.getAdminAttribute();
+            } else {
+                adminPrincipalAttribute = AZURE_LOGIN;
+            }
+            final Optional<String> username = Optional.ofNullable(user.getAttribute(adminPrincipalAttribute));
```
Comment (DementevNikita) [inline odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/auth/handler/impl/AzureUserHandler.java:53]: I'd not create an Optional here, simple `if` should do the trick

```java
final String username = user.getAttribute(adminPrincipalAttribute)
if (username != null && containsIgnoreCase(provider.getAdminPrincipals(), username)) {
    isAdmin = true;
}
```

**Review by damirabdul — APPROVED** [review]

File: odd-platform-api/src/main/resources/application.yml (hunk 2 above)
Comment (adyachenkoprov) [reply odd-platform-api/src/main/resources/application.yml]: @DementevNikita Ok. I'll do

**Pull request merged**